kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerState.java

```diff
@@ -328,6 +333,79 @@ public ConsumerCommittedResponse committed(ConsumerCommittedRequest request) {
     return response;
   }
 
+  /**
+   * Returns the beginning offset of the {@code topic} {@code partition}.
+   */
+  long getBeginningOffset(String topic, int partition) {
+    lock.lock();
+    try {
+      if (consumer == null) {
+        throw new IllegalStateException("KafkaConsumerState has been closed.");
+      }
+
+      Map<TopicPartition, Long> response =
+          consumer.beginningOffsets(singletonList(new TopicPartition(topic, partition)));
+
+      if (response.size() != 1) {
+        throw new IllegalStateException(
```

Comment:
don't immediately have an idea of a better alternative, but IllegalStateException seems a bit semantically wrong here. ideas?

Reply:
sounds good from the users point of view, so good enough for me.